### ISSUE TITLE
Feature request: un-deprecate the defringe module

### ISSUE THREAD

**Issue by abnormally-distributed** [darktable-org/darktable#9810]
**Is your feature request related to a problem? Please describe.**
I have found that the original defringe module at times produces better results than the newer chromatic abberations module (regardless of which color is set as the guide). This is especially true for cameras with sensors that use a non-standard color filter array (ie, the RGBE sensor of the Sony DSC-F828), where the raw chromatic abberations module doesn't work. 

**Describe the solution you'd like**
I still find the new chromatic abberations module useful, so I think a good option is to simply undeprecate defringe. I think the defringe module is particuarly better at handling axial (longitudinal) chromatic abberations and other abberations such as those caused by lens flare or other issues. 

**Alternatives**
Alternatively, defringe and chromatic abberations modules could be combined into one module that simply has the options for both modules combined into one menu.

I can provide pictures showing examples where defringe is still useful, if desired.

**Comment by Nilvus** [comment]
@rawfiner and @TurboGit: as you work on that, what's your view?

**Comment by TurboGit** [comment]
I have yet to find an example where the defringe module give better result than the new chromatic aberration. If such example can be given we may reconsider but probably better to enhance the new module that un-deprecating the old one.

**Comment by rawfiner** [comment]
I agree with TurboGit
Would be nice to have some example pictures please @abnormally-distributed

**Comment by abnormally-distributed** [comment]
I should have some time this evening to get some examples posted.

**Comment by abnormally-distributed** [comment]
Sorry guys, I had to get some work done on my laptop. I've got it back now, and will be updating this post with the promised example

**Comment by abnormally-distributed** [comment]
Alright, here's an example. https://imgur.com/a/too4ZBG

Notice that the combination of the CA module with green guide followed by defringed is best. It retains subtly more detail while better addressing the fringing compared to the others. It's very close to the defringed alone, however, note the reduction of fringing in the lines between the bricks.

For each panel in the example the parameters were adjusted to look the best I could get it for that setting. 

I can post other examples if anyone wants

**Comment by rawfiner** [comment]
Thank you!
I won't be able to take a look this week, but I will do so next week.

**Comment by johnny-bit** [comment]
@abnormally-distributed - can you post RAW file also? would be great for testing I think.

**Comment by tpinfold** [comment]
I too have seen some images which work better with the defringe module. It would be nice to keep it as an option. I appreciate maintaining all these modules may be an issue, but if it can be kept without causing work for the developers it would be good.

**Comment by abnormally-distributed** [comment]
> I agree with @johnny-bit , would be nice to have the raw please

I totally forgot about this thread (been busy with moving lately) until tpinfold commented. 

A raw file isn't really necessary, since the raw chromatic abberations module doesn't work with the DSC-F828. The issue I described is relevant to the newer chromatic abberations module. 

For now I just use rawtherapee on images where chromatic abberations isn't satisfactory (figure I'd start the habit now in anticipation of the defringe module being removed in darktable). Their defringe module allows to select specific hues for desaturating around edges. Affinity photo also has a decent defringe tool that also allows selection of hue. 

But here's an example, cropped in to show a particular area where blue fringing is notable. 

No defringing/CA correction
![defringe-none](https://user-images.githubusercontent.com/24922749/139603526-7dd05f59-e288-476f-9a1f-aa7beab716d2.PNG)

Defringe module
![defringe-defringe](https://user-images.githubusercontent.com/24922749/139603532-1f9dd573-9211-43cc-9765-0cc2f125de77.PNG)

Chromatic abberations module (blue, radius 13.0, strength 0.5, standard correction mode) **
![defringe-CA](https://user-images.githubusercontent.com/24922749/139603538-4d25f041-2046-4f62-98c6-92deee7dda39.PNG)

Chromatic abberations module (blue, radius 13.0, strength 0.5, standard correction mode, very large chromatic abberations checked)**
![defringe-CA-large](https://user-images.githubusercontent.com/24922749/139603608-4903c3ba-f10e-4a4c-8986-07ed8f6602ee.PNG)

** Note that the blue and red modes did not result in significant defringing of the blue edges. I selected the parameters that gave the most satisfactory results to show here.

So, the CA module makes things too smooth when set to achieve the most blue defringing. the sharpness is improved with very large turned on. Defringe, however, gives the best combination of sharpness and fringe correction. The defringing module seems to be better for the following circumstances in my experience:

- transverse chromatic abberation often resulting in blue/purple - green chromatic abberation towards the edges of the photo

- blue-purple fringing on dark edges , especially where there is a lot of contrast (ie, blue fringing on tree branches where the sunlight is bright behind the branches)

Both modules are useful, so I don't wish to see the CA module abandoned -- simply keeping both is good and reduces the need for me to use multiple programs in editing.
